### ISSUE TITLE
Zero downtime node pool changes

### ISSUE THREAD

**Issue by villesau** [terraform-google-modules/terraform-google-kubernetes-engine#1236]
### TL;DR

Gcloud has documentation on how to upgrade node pools without downtime: https://cloud.google.com/kubernetes-engine/docs/tutorials/migrating-node-pool

However, to my understanding there's no way to achieve that with this module without manual actions involved.

### Detailed design
Maybe node_pools configs could have

```markdown
zero_downtime_upgrades
```

 flag to support this? 
 
Or then it could be the default behaviour. In any case, zero downtime upgrades are desired and I'm not able to come up with a scenario where downtime would be desirable.

### Additional information

This is GCloud official documentation about it: https://cloud.google.com/kubernetes-engine/docs/tutorials/migrating-node-pool

**Comment by morgante** [comment]
If you use the `update-variant` versions of the module, it will recreate new node pools before deleting the old ones. This is the same thing that is done in the `gcloud` tutorial.

**Comment by villesau** [comment]
Thanks! I somehow had missed that. Have you considered making it the default? Sounds like a behaviour that is the most desired to me.
